### DISCUSSION TITLE
내게 딱 맞는 건강디저트 추천시스템 구현하기

### DISCUSSION THREAD

**Discussion by thrcle** [Pseudo-Lab/9th-builder#8] [plan]
## 프로젝트 소개

<p align="center"><img src="https://takuti.github.io/Recommendation.jl/latest/assets/images/user-item-matrix.png" width="" height="150"> <img src="https://takuti.github.io/Recommendation.jl/latest/assets/images/cf.png" width="" height="200"><p align="center">*이미지 출처: https://takuti.github.io/Recommendation.jl/latest/collaborative_filtering/</p> </p>

- 인터넷 쇼핑을 하다보면 놀랄정도로 내가 찾던 아이템을 잘 찾아줄 때도 있고, 반대로  전혀 취향이 아닌 아이템을 추천줄 때도 있습니다.  ‘내 마음에 드는 상품을 추천 받으려면 어떻게 해야할까요?
- 평소 디저트를 좋아하긴하지만 잘못 먹으면 소화가 잘 안되어 나에게 맞는 성분의 디저트를 찾고 싶었고 이에 건강디저트 플랫폼에서 직접 구현해보자는 도전의식이 생겼습니다. (건강디저트 플랫폼 구경하러 가기➡️[빵그리의 오븐](https://www.bbangle.store/) *출시 준비중)
- 구현한 추천 시스템을 실제 적용까지 하는 것을 목표로 하고 있기 때문에 결과물에 대한 성취감을 느낄 수 있습니다. 
- 수도콘에서 결과물에 대한 데모부스까지 해볼 계획입니다. 목표가 확실한만큼 짧은시간내에 책임감있고 밀도있는 프로젝트가 될 것입니다. 목표를 가지고 함께 불태워보실까요?🔥 

### 프로젝트 목표
1. 주요 추천 시스템 이론에 대한 구현 방식 이해(각 방식의 원리, 특징, 장단점 파악)
2. 학습한 내용을 기반으로 실전 프로젝트에 적용: 추천시스템 구현 및 반영까지 

### 프로젝트 내용
Part 0. OT 및 스터디 목표 안내 (1주)
    - 자기소개 
    - 진행일정 및 프로젝트 목표 공유 
    
 Part 1. 추천시스템 학습 (4주) : 알고리즘 학습 및 관련 사례 탐색
    - 협업 필터링 방식 
    - Matrix Factorization(MF)
    - 딥러닝
    - 하이브리드 추천 시스템 학습
    
 Part 2. 실전 적용 (8주)
    - 추천 시스템 구조 & 모델링 전략 설계
 	- 데이터 수집 및 전처리
  	- 이론 적용 및 피드백 및 보완

## 모집 대상
**지원 대상**
- ${\textsf{\color{red}스터디를 끝까지 함께할 의지/시간적 여유(불참 2회 이내)가 있으신 분}}$ 
- 추천시스템에 대한 관심이 있고, 직접 구현-적용까지 해보고싶으신 분
- Python/추천시스템에 대한 기초 지식이 있으신 분

**우대** 
- 추천시스템을 구현한 경험이 있으신 분

**모집 인원**
- 4~5명

### 첫 모임 날짜 & 시간 & 장소
- 날짜: 9/12 목
- 시간: 오후 9시 (약 1시간 소요)
- 장소: 온라인 디스코드

### 정기 모임 시간 & 장소

- 시간: 매주 목요일 오후 9시-11시
- 장소: 온라인 디스코드

### 학습 참고 자료
- [python을 이용한 개인화 추천 시스템 제 2판 (임일, 도서출판 청람)](https://m.yes24.com/Goods/Detail/110328538 )
<img src="https://image.yes24.com/goods/110328538/XL.png" width="150" height="200"/>
- 이론 스터디에서 사용할 실습 코드는 도서출판 청람 홈페이지>일반자료실에서 다운 가능합니다. 

### 스터디 활동 방법
- **오픈 아카데미** 형태(=청강 가능)로 진행하며, 발표자료는 가짜연구소를 통해 **공개적으로 배포할** 수 있도록 할 예정입니다
- **3회** 이상 불참 시  프로젝트 러너에서 제외됩니다.
    
1. **진행 & 발표**
- 전체리딩은 빌더가 하되, 인원에 맞게 주차별 발표자를 분배하여  간단하게 주제 정리 하는 발표시간을 가질 예정입니다.(참여자 모두 최소 1번씩은 발표할 수 있도록)
    - 출석체크를 위해 **실명**으로 참여 부탁드립니다.
    - 집중력있는 분위기를 위해 **카메라는 ON**으로 진행합니다.   
    - 발표자는 OT 날 선정 (참여 인원에 따라 1인 1-2회 정도 발표 예상)
- 크게 part1 스터디, part2실전 적용으로 나눠 진행 예정.
- Part1의 경우 각 주차에 해당하는 주제를 스터디한 후 정리/어려웠던 점/논의하고 싶은 부분 토의. 
- Part2의 경우 건강디저트 개인화 추천시스템을 구현해보고 성능평가&상호 피드백 및 적용

    
2. **학습 활동 (필수)**
    - 주차별 학습내용 Jupyter Book 형태로 정리
    - **학습 내용에 대한 질문 및 공유**

## 주차별 목표와 학습 내용
💡 주차별 진행계획은 러너(스터디원)분들과 협의에 의해 변경될 수 있습니다.

| 날짜 | 내용 | 발표자 | 
| ----- | ----- | -------- |
| 2024/09/12 | OT (OT 및 스터디 목표 안내,진행일정 및 프로젝트 목표 공유) | 심세원| 
| 2024/09/19 |  Part 1. 추천시스템 학습:협업 필터링 방식 &사례탐색| 미정| 
| 2024/09/26|  Part 1. 추천시스템 학습:Matrix Factorization(MF)&사례탐색 | 미정| 
| 2024/10/03|  Part 1. 추천시스템 학습:딥러닝 &사례탐색| 미정| 
| 2024/10/10|  Part 1. 추천시스템 학습:하이브리드 추천 시스템 학습 &사례탐색| 미정| 
| 2024/10/17|  Part2. 데이터 수집&전처리 프로세스 이해|| 미정| 
| 2024/10/24|  Part2. 추천 시스템 구조 & 모델링 전략 설계| 미정| 
| 2024/10/31|  Part2. 추천 시스템 구조 & 모델링 전략 설계| 미정| 
| 2024/11/7|  Part2. 모델링 실습 공유&피드백| 미정| 
| 2024/11/14|  Part2. 모델링 실습 공유&피드백| 미정| 
| 2024/11/21|  Part2. 모델링 실습 공유&피드백| 미정| 
| 2024/11/28|  Part2. 모델 배포| 미정| 
| 2024/12/05|  Part2. 추후업데이트 예정| 미정| 
| 2024/12/12|  Part2. 추후업데이트 예정| 미정|

**Comment by WongiPark0628** [comment]
유익해보여 기대됩니다..!

**Reply by lynnminn** [reply]
스터디 제목부터 엄청 재밌어보입니다 ㅎㅎ 추천 시스템 기본적인 내용부터 프로젝트까지 진행하면 공부할 때도 확 와닿을 것 같아요!!
